rectools/models/rank.py

```diff
@@ -0,0 +1,181 @@
+#  Copyright 2022 MTS (Mobile Telesystems)
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Implicit ranker model."""
+
+import typing as tp
+from enum import Enum
+
+import implicit.cpu
+import numpy as np
+from implicit.cpu.matrix_factorization_base import _filter_items_from_sparse_matrix as filter_items_from_sparse_matrix
+from scipy import sparse
+
+from rectools import InternalIds
+from rectools.models.base import Scores
+
+
+class Distance(Enum):
+    """Distance metric"""
+
+    DOT = 1  # Bigger value means closer vectors
+    COSINE = 2  # Bigger value means closer vectors
+    EUCLIDEAN = 3  # Smaller value means closer vectors
+
+
+class ImplicitRanker:
+    """
+    Ranker model which uses implicit library matrix factorization topk method
```

Comment:
Let's expand here: "This ranker is suitable for the following cases of scores calculation: `subject_embeddings.dot(objects_embeddings)` and `subject_interactions.dot(item-item-similarities)`"